File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        
        int result = 0, zeros_count = 0;
        for(int l = 0, r = 0; r < nums.size(); r++){
            zeros_count += nums[r] == 0? 1: 0;
            while(zeros_count == 2){
                zeros_count -= nums[l] == 0? 1: 0;
                l++;
            }
            result = max(result, r - l + 1 - zeros_count);
        }
        return result == nums.size()? result - 1: result;
    }
};

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        zeroes = 0
        l = 0
        longest = 0
        for r in range(len(nums)):
            zeroes += 1 - nums[r]
            while l <= r and zeroes > 1:
                zeroes -= 1 - nums[l]
                l += 1
            
            longest = max(longest, r - l)
        
        return longest